models/surveys.go

```diff
@@ -486,15 +493,46 @@ func (api *API) Info(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// AllSurveys returns summaries of all known surveys. The surveys are returned in ascending short name order.
+// AllSurveys returns a list of all known surveys
 func (api *API) AllSurveys(w http.ResponseWriter, r *http.Request) {
-	rows, err := api.AllSurveysStmt.Query()
+	var rows *sql.Rows
+	var err error
+	rows, err = api.AllSurveysStmt.Query()
+	if err != nil {
+		logError("Get all surveys returned error", err)
+		http.Error(w, "Failed to retrieve surveys", http.StatusInternalServerError)
+		return
+	}
+	parseSurveys(rows, w)
+}
 
+// SurveysByType returns surveys of a particular type
+func (api *API) SurveysByType(w http.ResponseWriter, r *http.Request) {
+	var rows *sql.Rows
+	var err error
+
+	urlSplit := strings.Split(r.URL.String(), "/")
+	surveyType := urlSplit[len(urlSplit)-1]
+
+	// Valid types are enumerated in DB
+	if surveyType != "Business" && surveyType != "Social" && surveyType != "Census" {
+		logError("Invalid surveyType in SurveysByType", fmt.Errorf("surveyType:%s", surveyType))
+		http.Error(w, "Failed to retrieve surveys", http.StatusBadRequest)
+		return
+	}
+
+	rows, err = api.GetSurveysBySurveyTypeStmt.Query(surveyType)
 	if err != nil {
-		http.Error(w, "AllSurveys query failed", http.StatusInternalServerError)
+		logError("Get surveys by type returned error", err)
+		http.Error(w, "Failed to retrieve surveys", http.StatusInternalServerError)
 		return
 	}
+	parseSurveys(rows, w)
+}
 
+// parseSurveys extracts and returns survey data
```

Comment:
Although you can for completeness, you don't need to put comments onto private functions in Go as they won't get parsed into the generated `godoc`